scripts/add_brownfield.py

```diff
@@ -75,16 +76,22 @@ def add_brownfield(n, n_p, year):
         for tattr in n.component_attrs[c.name].index[selection]:
             n.import_series_from_dataframe(c.pnl[tattr], c.name, tattr)
 
+        # deal with gas network
+        pipe_carrier = ['gas pipeline']
+        to_drop = n.links.carrier.isin(pipe_carrier) & (n.links.build_year!=year)
```

Comment:
Is that correct? It removes all gas pipelines not build in the exact year? Or am I reading this wrong?

Reply:
Good quesiton. I added this to avoid duplicating the capacity of the existing gas network for each investment period.  I think with the retrofitting of the gas pipes it does make sense. Without the option to retrofit the gas pipelines their capacity can be expanded further than the current capacity. It maybe makes sense then to keep the gas pipes from the previous investment period with their optimized capacity and with `p_nom_extendable=False` and to set the capacity of all the "new build" gas pipes of the current investment period to `p_nom=0`. I make a suggestion!